crates/wasi-component-loader/src/extension/api/since_0_14_0/wit/grpc/client.rs

```diff
@@ -0,0 +1,140 @@
+use super::{
+    ClientConfiguration, GrpcClient, GrpcStatus, GrpcStreamingResponse, GrpcUnaryResponse, HostGrpcClient, MetadataMap,
+};
+use crate::WasiState;
+use bytes::BufMut as _;
+use dashmap::Entry;
+use tonic::metadata::{MetadataKey, MetadataValue};
+use wasmtime::component::Resource;
+
+impl HostGrpcClient for WasiState {
+    async fn new(
+        &mut self,
+        configuration: ClientConfiguration,
+    ) -> wasmtime::Result<Result<Resource<GrpcClient>, String>> {
+        let uri: http::Uri = match configuration.uri.parse() {
+            Ok(uri) => uri,
+            Err(err) => return Ok(Err(err.to_string())),
+        };
+
+        let client = match self.grpc_clients().entry(configuration.uri) {
```

Comment:
Not a fan of the configuration (apart from the url) getting ignored if the client is already initialized.